### PR TITLE
Enable HTTP GET/POST adaptors to specify request headers

### PR DESCRIPTION
**Pull request by Empact**
https://www.pivotaltracker.com/story/show/164028351

### CODE REVIEW

File: adapters/http.go [hunk 1]
```diff
@@ -3,27 +3,31 @@ package adapters
...
-// HTTPGet requires a URL which is used for a GET request when the adapter is called.
-type HTTPGet struct {
-	URL models.WebURL `json:"url"`
-	GET models.WebURL `json:"get"`
-}
-
-// Perform ensures that the adapter's URL responds to a GET request without
-// errors and returns the response body as the "value" field of the result.
-func (hga *HTTPGet) Perform(input models.RunResult, _ *store.Store) models.RunResult {
+func request(input models.RunResult, method, url string, headers http.Header, body io.Reader, contentType string) models.RunResult {
```
Comment (j16r) [inline adapters/http.go]: I'd move this private method to lower in the file, public stuff should go at the top IMO.

File: integration/features_test.go [hunk 2]
```diff
@@ -40,6 +40,69 @@ func TestIntegration_Scheduler(t *testing.T) {
...
+	newHeads <- models.BlockHeader{Number: cltest.BigHexInt(safe)} // 23465
+	eth.EventuallyAllCalled(t)
+	cltest.WaitForTxAttemptCount(t, app.Store, 1)
+
+	jr = cltest.WaitForJobRunToComplete(t, app.Store, jr)
+
+	eth.EventuallyAllCalled(t)
+}
+
+func TestIntegration_FeeBump(t *testing.T) {
```
Comment (j16r) [inline integration/features_test.go]: 👍

File: integration/features_test.go [hunk 3]
```diff
@@ -40,6 +40,69 @@ func TestIntegration_Scheduler(t *testing.T) {
 }
 
 func TestIntegration_HelloWorld(t *testing.T) {
+	tickerHeaders := http.Header{
```
Comment (j16r) [inline integration/features_test.go]: Got a better name for HelloWorld?

File: adapters/http.go (hunk 1 above)
Comment (Empact) [reply adapters/http.go]: Moved to the bottom

File: integration/features_test.go (hunk 3 above)
Comment (Empact) [reply integration/features_test.go]: now: `HttpRequestWithHeaders`

**Comment by Empact** [comment]
Addressed feedback

**Pull request merged**